test/e2e/storage/testsuites/multiattach.go

```diff
@@ -0,0 +1,371 @@
+			nil, node1, framework.PodStartTimeout)
+		defer func() {
+			framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod1))
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating 2nd pod with a volume")
+		pod2, err := framework.CreateSecPodWithNodeName(l.cs, l.ns.Name,
+			[]*v1.PersistentVolumeClaim{l.resource2.pvc},
+			false, "", false, false, framework.SELinuxLabel,
+			nil, node2, framework.PodStartTimeout)
+		defer func() {
+			framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod2))
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Checking if the volume in 1st pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod1, pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the volume in 1st pod works properly")
+		utils.CheckReadWriteToPath(pod1, pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if the volume in 2nd pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod2, pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the volume in 2nd pod works properly")
+		utils.CheckReadWriteToPath(pod2, pattern.VolMode, "/mnt/volume1")
+
+		// Delete pod1
+		framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod2))
+
+		// Test read/write work well in pod1 after pod2 is deleted
+		By("Rechecking if the volume in 1st pod exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod1, pattern.VolMode, "/mnt/volume1")
+
+		By("Rechecking if read/write to the volume in 1st pod works properly")
+		utils.CheckReadWriteToPath(pod1, pattern.VolMode, "/mnt/volume1")
+	}
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- same volume mode
+	//   [volume1]  [volume2]
+	It("should attach two volumes with the same volume mode to the same pod", func() {
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = l.resource1.pattern.VolMode
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- different volume mode (only <block, filesystem> pattern is tested)
+	//   [volume1]  [volume2]
+	It("should attach two volumes with different volume mode to the same pod", func() {
+		if pattern.VolMode == v1.PersistentVolumeFilesystem {
+			framework.Skipf("Filesystem volume case should be covered by block volume case -- skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		// Only Block volume test for 1st volume will reach here,
+		// so we should set filesystem for 2nd volume.
+		l.resource2.pattern.VolMode = v1.PersistentVolumeFilesystem
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	// [pod1] [pod2]
+	// [   node1   ]
+	//   \      /     <- same volume mode
+	//   [volume1]
+	It("should attach the same volume with the same volume mode to the different pod on the same node", func() {
+		init()
+		defer cleanup()
+
+		l.resource2.pvc = l.resource1.pvc
+		testAttachOneVolumeToTwoPods(l, true /* sameNode */)
```

Comment:
the last 2 tests will work with preprovisionedPVTestDrivers right? is there value in testing them, besides iscsi? If so, we can wrap the above two in "if is a DynamicPVTestDriver". However, then the testsuite accepted patterns and the preconditions will also need to be changed.

Reply:
Good point. I will add codes to skip pre-provisioned volume for tests that require two volumes, then add pre-provisioned volume to the list of test patterns to execute. Then, what we originally would like to test can be tested. 

> is there value in testing them, besides iscsi?

I guess that this issue won't be iscsi specific, so testing it for other drivers will have benefit to all drivers.